libs/evcs-service/src/main/java/uk/gov/di/ipv/core/library/service/EvcsService.java

```diff
@@ -139,26 +148,51 @@ private void persistEvcsUserVCs(
                                             new EvcsUpdateUserVCsDto(
                                                     getVcSignature(vc), EvcsVCState.HISTORIC, null))
                             .toList();
-            var existingPendingReturnEvcsUserVcsToUpdate =
+            var existingPendingReturnUserVcsNotInSessionToUpdate =
                     existingEvcsUserVCs.stream()
                             .filter(vc -> vc.state().equals(PENDING_RETURN))
+                            .filter(
+                                    vc ->
+                                            credentials.stream()
+                                                    .noneMatch(
+                                                            credential ->
+                                                                    credential
+                                                                            .getVcString()
+                                                                            .equals(vc.vc())))
                             .map(
                                     vc ->
                                             new EvcsUpdateUserVCsDto(
                                                     getVcSignature(vc),
                                                     EvcsVCState.ABANDONED,
                                                     null))
                             .toList();
-            var evcsUserVCsToUpdate =
+            var existingPendingReturnUserVcsInSessionToUpdate =
+                    existingEvcsUserVCs.stream()
+                            .filter(vc -> vc.state().equals(PENDING_RETURN))
+                            .filter(
+                                    vc ->
+                                            credentials.stream()
+                                                    .anyMatch(
+                                                            credential ->
+                                                                    credential
+                                                                            .getVcString()
+                                                                            .equals(vc.vc())))
+                            .map(
+                                    vc ->
+                                            new EvcsUpdateUserVCsDto(
+                                                    getVcSignature(vc), EvcsVCState.CURRENT, null))
```

Comment:
N.B. the CURRENT and HISTORIC updates should only happen when storing current identities (not pending ones). I don't think this can actually happen in practice, but might be worth adding a guard?

(If we're storing a pending identity, it's just the pending -> abandoned that we care about)